pixi/src/checks/LinterCheck.js

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 The AMPHTML Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const API_ENDPOINT = API_ENDPOINT_LINTER;
+
+export default class LinterCheck {
+  constructor() {
+    this.apiUrl = new URL(API_ENDPOINT);
+  }
+
+  async run(pageUrl) {
+    this.apiUrl.searchParams.set('url', pageUrl);
+
+    try {
+      const apiResult = await this.fetchJson();
+      return this.createReportData(null, apiResult);
+    } catch (e) {
+      return this.createReportData(e);
+    }
+  }
+
+  createReportData(error, apiResult) {
```

Comment:
I'd split this into two different methods: `createReport` and `createError`. This simplifies the parameters and you avoid the additional if check. We could do the same for all the checks.

Reply:
Implemented for this one for now